File: code.py
"""
March 25, 2025
by: Sophie Nguyen
 
using sonar with pico

"""

import time
import board
import adafruit_hcsr04
import digitalio


# variables
seconds_to_microseconds_conversion_number = 1000000
sonar_delays = [2 / seconds_to_microseconds_conversion_number, 10 / seconds_to_microseconds_conversion_number]
delay_between_sonar_cheeks = 1
distance = 0
TOO_CLOSE = 20

# setup
led = digitalio.DigitalInOut(board.GP12)
led.direction = digitalio.Direction.OUTPUT
sonar = adafruit_hcsr04.HCSR04(trigger_pin = board.GP15, echo_pin = board.GP14)

# loop
while True:
    # Sonar gets the distance form object
    time.sleep(delay_between_sonar_cheeks)
    distance = sonar.distance

    print(f"Distance: {distance} cm")

    # Turns on LED if an object’s distance is equal to or closer then 20 cm from the sonar
    if distance <= TOO_CLOSE:
        led.value = True
    else:
        led.value = False

    # The commented out code is not part of the actual code but is needed to get it working by uncommenting it and then recommenting it
    #time.sleep(delay_between_sonar_cheeks)

File: code.c

/*
 *
 *March 25, 2025
 * by: Sophie Nguyen
 * 
 * using sonar with arduino
 * 
 */

//pin numbers
const int TRIG_PIN = 3;  
const int ECHO_PIN = 2; 
const int LED = 11;
const int SPEED_OF_LIGHT = 0.0343;


//variables
float duration, distance;  

//setup
void setup() {  
   pinMode(TRIG_PIN, OUTPUT);  
   pinMode(ECHO_PIN, INPUT); 
   pinMode(LED, OUTPUT); 
   Serial.begin(9600);  
}  

//loop
void loop() {  
  //clears the trig pin
   digitalWrite(TRIG_PIN, LOW);  
   delayMicroseconds(2);  

   //sets trig pin on high fpr 10 micro seconds
   digitalWrite(TRIG_PIN, HIGH);  
   delayMicroseconds(10);  
   digitalWrite(TRIG_PIN, LOW);  

   //reads the echo pin, returns the sound wave travel time in microseconds
   duration = pulseIn(ECHO_PIN, HIGH);  

   distance = (duration*SPEED_OF_LIGHT)/2;

   if (distance < 10) {
    digitalWrite(LED, HIGH);
   }

   else {
     digitalWrite(LED,LOW);
   }

    Serial.print(distance);

    Serial.println(" cm");

    delay(100);
}